sklearn/neighbors/nearest_centroid.py

```diff
@@ -133,6 +133,8 @@ def fit(self, X, y):
                     self.centroids_[cur_class] = np.median(X[center_mask], axis=0)
                 else:
                     self.centroids_[cur_class] = csc_median_axis_0(X[center_mask])
+            elif self.metric == 'precomputed':
```

Comment:
You can raise it as early as here (https://github.com/sergulaydore/scikit-learn/blob/38ad1d24690fea9f8d1867c2f8b9587c0204268d/sklearn/neighbors/nearest_centroid.py#L98), with an explicit error message saying `metric="precomputed"` not supported.